libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/controller/RoutePreviewStateController.kt

```diff
@@ -109,48 +121,63 @@ class RoutePreviewStateController(private val store: Store) : StateController()
             }
     }
 
-    private suspend fun MapboxNavigation.fetchRoute(
-        routeOptions: RouteOptions,
-        fetchStarted: (requestId: Long) -> Unit
-    ): List<NavigationRoute> {
-        return suspendCancellableCoroutine { cont ->
-            val requestId = requestRoutes(
-                routeOptions,
-                object : NavigationRouterCallback {
-                    override fun onRoutesReady(
-                        routes: List<NavigationRoute>,
-                        routerOrigin: RouterOrigin
-                    ) {
-                        cont.resume(routes)
-                    }
-
-                    override fun onFailure(
-                        reasons: List<RouterFailure>,
-                        routeOptions: RouteOptions
-                    ) {
-                        cont.resumeWithException(FetchRouteError(reasons, routeOptions))
+    // This SAGA dispatches FetchRoute action, awaits RoutePreviewState.Ready
+    // and updates NavigationState to either RoutePreview or ActiveNavigation state.
+    private suspend fun fetchRouteAndMoveToNavigationStateSaga() {
+        store.actionsFlowable()
+            .filter {
+                it is RoutePreviewAction.FetchRouteAndShowRoutePreview ||
+                    it is RoutePreviewAction.FetchRouteAndStartActiveNavigation ||
+                    it is NavigationStateAction.Update
```

Comment:
Similar to above, why are we filtering `it is NavigationStateAction.Update` if we are not handling it when we `collectLatest`?

Reply:
Before, fetch route thunk actions were bound to `StartNavigationButtonComponent` and `RoutePreviewButtonComponent`'s `CoroutineScopes`, and were cancelled when those components were detached.

Now, these operations are bound to `RoutePreviewStateController`'s `CoroutineScope`, which is cancelled only when `MapboxNavigationApp.lifecycleOwner` is destroyed. So, to ensure the in-flight route requests are cancelled, we need to listen to any `NavigationStateAction.Update` actions.